pyaerocom/helpers.py

```diff
@@ -204,7 +205,12 @@ def tuple_list_to_lists(tuple_list):
     return list(map(list, zip(tuple_list)))
 
 
-def make_dummy_cube_latlon(lat_res_deg=2, lon_res_deg=3, lat_range=None, lon_range=None):
+def make_dummy_cube_latlon(
+    lat_res_deg: float = 2,
+    lon_res_deg: float = 3,
+    lat_range: Optional[Sequence[float]] = None,
+    lon_range: Optional[Sequence[float]] = None,
```

Comment:
The function can also accept lat_range and lon_range as lists. Will type-annotating them to just be tuples fail? At first I had something similar with Tuple, but think it caused issues. I couldn't figure out how to type-annotate it to accept either a tuple or a list,l but with only two elements.

Reply:
> The function can also accept lat_range and lon_range as lists. Will type-annotating them to just be tuples fail?


In general type annotations do not have runtime effects. There are some packages that use types to define runtime behaviour, such as Pydantic and Typer, but those are exceptions.

> At first I had something similar with Tuple, but think it caused issues. I couldn't figure out how to type-annotate it to accept either a tuple or a list,l but with only two elements.


List are mutable objects, so the most that a type annotation can do is to restrict the type to list of floats of any length with `list[float]`. Tuples are immutable objects, so the typing annotations can be more restrictive. Here I proposed a tuple of 2 floats with `tuple[float, float]`, but a tuple of any lenght can be annotated with `tuple[float, ...]`.

If the function accepts a list, it would be wise to place a guarding clause at the top of the function that assert that a list has 2 elements.